packages/atlas/src/providers/assets/operatorsProvider.tsx

```diff
@@ -50,17 +54,31 @@ export const OperatorsContextProvider: FC<PropsWithChildren<unknown>> = ({ child
   const [distributionOperatorsError, setDistributionOperatorsError] = useState<unknown>(null)
   const [storageOperatorsError, setStorageOperatorsError] = useState<unknown>(null)
   const [failedStorageOperatorIds, setFailedStorageOperatorIds] = useState<string[]>([])
+  const {
+    coordinates,
+    expiry,
+    actions: { setUserLocation },
+  } = useUserLocationStore((state) => state)
 
   const client = useApolloClient()
 
-  const fetchDistributionOperators = useCallback(() => {
+  const fetchDistributionOperators = useCallback(async () => {
+    const now = new Date()
+    let userCoordinates: Coordinates
     const distributionOperatorsPromise = client.query<
       GetDistributionBucketsWithOperatorsQuery,
       GetDistributionBucketsWithOperatorsQueryVariables
     >({
       query: GetDistributionBucketsWithOperatorsDocument,
       fetchPolicy: 'network-only',
     })
+    if (!coordinates || !expiry || now.getTime() > expiry) {
+      const getUserCoordinates = await axios.get(USER_LOCATION_SERVICE)
```

Comment:
It would be good to handle error here in some way too. You know, if the service was temporary unavailable or something